daml/Marketplace/Trading/Service.daml

```diff
@@ -25,8 +31,46 @@ template Service
     key (operator, provider, customer) : (Party, Party, Party)
     maintainer key._1
 
+    let
+      withdrawFromAllocationAccount : Order.TradeCollateral -> Update Order.TradeCollateral
+      withdrawFromAllocationAccount = \tc -> case tc of
+        (Order.Cleared party)         -> return $ Order.Cleared party
+        (Order.Collateral depositCid) -> Order.Collateral <$> exerciseByKey @AllocationAccountRule.T allocationAccount.id AllocationAccountRule.Withdraw with transferTo = tradingAccount, ..
+
     controller customer can
-      nonconsuming RequestCreateOrder : (ContractId Order.T, ContractId CreateOrderRequest, ContractId AssetDeposit)
+      nonconsuming RequestCreateClearedOrder : Either (ContractId Order.T)
+                                                      (ContractId Order.T, ContractId CreateOrderRequest, Order.TradeCollateral)
+        with
+          clearinghouse : Party
+          details : Order.Details
+        do
+          -- TODO: Check that price and quantity conform to the precision requirements of the Listing
+          -- TODO: Check order valid for exberry with the market/limit and timeInForce combos (also, if GTD if expiry date is less than one year)
+          let
+            status = Order.New
+            providerOrderId = None
+            executions = []
+            remainingQuantity = details.asset.quantity
+            collateral = Order.Cleared clearinghouse
+
+          optClearingService <- lookupByKey @Clearing.Service (operator,clearinghouse,customer)
+
+          case optClearingService of
+            None -> do
+              let reason = Error with code = "1234"; message = "Customer is not a member of clearinghouse"
```

Comment:
Will we want to create different error codes for different errors?

Reply:
I think the initial reason for having error codes comes from Exberry returning error codes - so it could be good to do keep track of their codes somehow, and add our own. Looks like all of exberry's codes are in the 1000s so the marketplace model specific ones could be in the 3000s or something to that effect